internal/graph/graph.go

```diff
@@ -420,6 +421,33 @@ func newTree(prof *profile.Profile, o *Options) (g *Graph) {
 	return selectNodesForGraph(nodes, o.DropNegative)
 }
 
+var javaFullRegExp = regexp.MustCompile(`^(?:[a-z]\w*\.)*[A-Z][\w\$]*\.(?:<init>|[a-z]\w*(?:\$\d+)?)(?:\(|$)`)
+var javaShortRegExp = regexp.MustCompile(`([A-Z][\w\$]*\.(?:<init>|[a-z]\w*(?:\$\d+)?))(?:|$)`)
+var goPrefixRegExp = regexp.MustCompile(`^(?:[a-z]\w*(\.\w*)*\/)+.`)
+var goPathRegExp = regexp.MustCompile(`^(?:[a-z]\w*(\.\w*)*\/)+`)
+var cppFullRegExp = regexp.MustCompile(`^(?:[_a-zA-Z]\w*\::)*_*[A-Z]\w*::~?[_a-zA-Z]\w*$`)
+var cppAnonymousPrefixRegExp = regexp.MustCompile(`^\(anonymous namespace\)::`)
+
+// GetNodeShortName returns a shortened version of a function's name.
+func GetNodeShortName(f string) string {
+	if goPrefixRegExp.MatchString(f) {
```

Comment:
How confident are we that we would have only a single match? Are there any names that could potentially match go, java and C++?

Reply:
I modified this function somewhat.

The go regexp (goPrefixRegExp), java regexp (javaFullRegExp), and cpp regexp () must include the beginning of the string, and java and cpp match to the end of the string. I think a function must have a "/" to match go, it must have a "::" to match c++ (and cannot have a "/"), and java can't have either a "/" or a "::"

So I'm fairly confident only one of these can ever be matched.